src/dbus.c

```diff
@@ -200,6 +201,19 @@ static void on_notify(GDBusConnection * connection,
                                         if (dict_value)
                                                 raw_icon = get_raw_image_from_data_hint(dict_value);
 
+                                        /* Check for transient hints
+                                         *
+                                         * According to the spec, the transient hint should be boolean.
+                                         * But notify-send does not support hints of type 'boolean'.
+                                         * So let's check for int and boolean until notify-send is fixed.
+                                         */
+                                        if((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_BOOLEAN)))
+                                                transient = g_variant_get_boolean(dict_value);
+                                        if((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_UINT32)))
```

Comment:
yes. that's good. I've always wondered, why you gave me this snippet:

```
if ((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_UINT32)))
```

With else if constructs this makes totally sense.

btw: updated it.